src/dbscan.jl

```diff
@@ -132,15 +132,15 @@ points = randn(3, 10000)
 clusters = dbscan(points, 0.05, min_neighbors = 3, min_cluster_size = 20) # clusters with less than 20 points will be discarded
 ```
 """
-function dbscan{T <: AbstractFloat, N}(points::Array{T, N}, radius::Real; leafsize::Int = 20, kwargs ...)
+function dbscan{T <: Real}(points::AbstractMatrix{T}, radius::T; leafsize::Int = 20, kwargs ...)

```

Comment:
I think `radius::T` is too restrictive, and it would be nice to support different types for `points` and `radius`.
In fact, it's also too restrictive in `_dbscan()` (which is the cause of #84).
There, `radius` is compared with the result of `NearestNeigbours.inrange()`, which might have different type than `T`.
So I'd rather set it to `::Real` in both methods.

Reply:
I believe `Real` for radius is quite sufficient, it's a positive real number. For points, the parameterization can be drop.